internal/cmd/egctl/translate.go

```diff
@@ -195,58 +209,139 @@ func translate(w io.Writer, inFile, inType, outType, output, resourceType string
 		return fmt.Errorf("unable to read input file: %w", err)
 	}
 
-	if inType == gatewayAPIType && outType == xdsType {
-		return translateFromGatewayAPIToXds(w, inBytes, output, resourceType, addMissingResources)
+	if inType == gatewayAPIType {
+		var toXds, echoGatewayAPI bool
+		for _, outType := range outTypes {
+			if outType == xdsType {
+				toXds = true
+			} else if outType == gatewayAPIType {
+				echoGatewayAPI = true
+			}
+		}
+
+		// Unmarshal input
+		gcName, resources, err := kubernetesYAMLToResources(string(inBytes), addMissingResources)
+		if err != nil {
+			return fmt.Errorf("unable to unmarshal input: %w", err)
+		}
+
+		// Translate from Gateway API to Xds IR
+		gTranslator := &gatewayapi.Translator{
+			GatewayClassName:       v1beta1.ObjectName(gcName),
+			GlobalRateLimitEnabled: true,
+		}
+		gRes := gTranslator.Translate(resources)
+
+		results := []interface{}{}
+		if echoGatewayAPI {
+			results = append(results, resources)
+		}
+
+		if toXds {
+			keys := []string{}
+			for key := range gRes.XdsIR {
+				keys = append(keys, key)
+			}
+			// Make output stable
+			sort.Strings(keys)
```

Comment:
imo it should be inherently stable, we have `sort` inside the gateway-api lib, is that not the case ?

Reply:
The map iteration after the gateway-api translation doesn't have a stable order:
https://github.com/envoyproxy/gateway/blob/384577334dea2c0ba28800643a2b94fea4eed244/internal/cmd/egctl/translate.go#L220

Reply:
thanks for clarifying that, can you add that as a comment - `Make output stable since XdsIR is a map`